File: solutions/go/blackjack/1/blackjack.go
package blackjack

func ParseCard(card string) int {
    var parsedCard int
    switch card{
        case "ace": parsedCard = 11	
        case "two": parsedCard = 2	
        case "three": parsedCard = 3	
        case "four": parsedCard = 4	
        case "five": parsedCard = 5	
        case "six": parsedCard = 6	
        case "seven": parsedCard = 7	
        case "eight": parsedCard = 8
        case "nine": parsedCard = 9
        case "ten": parsedCard = 10
        case "jack": parsedCard = 10
        case "queen": parsedCard = 10
        case "king": parsedCard = 10
        default: parsedCard = 0   
    }
	return parsedCard
}

func FirstTurn(card1, card2, dealerCard string) string {
    var action string
	switch{
        case cardsSum(card1, card2) == 22:
    		action = "P"
        case cardsSum(card1, card2) == 21 && ParseCard(dealerCard) < 10:
    		action = "W"
        case  17 <= cardsSum(card1, card2):
    		action = "S"
        case  12 <= cardsSum(card1, card2) && ParseCard(dealerCard) < 7 :
    		action = "S"
        default: action = "H"
    }
	return action
}

func cardsSum(card1, card2 string) int{
    return ParseCard(card1) + ParseCard(card2)
}

File: solutions/ruby/blackjack/1/blackjack.rb
module Blackjack
  def self.parse_card(card)
    case card
      when "ace" then 11	
      when "two" then 2	
      when "three" then 3	
      when "four" then 4	
      when "five" then 5	
      when "six" then 6	
      when "seven" then 7	
      when "eight" then 8
      when "nine" then 9
      when "ten" then 10
      when "jack" then 10
      when "queen" then 10
      when "king" then 10
      else 0
    end
  end

  def self.card_range(card1, card2)
    case parse_card(card1) + parse_card(card2)
      when (4..11) then "low"
      when (12..16) then "mid"
      when (17..20) then "high"
      when 21 then "blackjack"
      else "to-high"
    end
  end

  def self.first_turn(card1, card2, dealer_card)
    case card_range(card1, card2)
    when "to-high" then "P"
    when "high" then "S"
    when "mid" then parse_card(dealer_card) >= 7 ? "H" : "S"
    when "low" then "H"
    when "blackjack" then parse_card(dealer_card)< 10 ? "W" : "S"
    end
  end
end
